src/Kirel.Repositories/KirelGenericEntityFrameworkRepository.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System.Linq.Expressions;
+using Kirel.Repositories.Interfaces;
+using Kirel.Repositories.Sorts;
+using Kirel.Shared;
+using Microsoft.EntityFrameworkCore;
+
+namespace Kirel.Repositories.Infrastructure.Generics;
+
+/// <summary>
+/// Generalized Repository class type 
+/// </summary>
+public class KirelGenericEntityFrameworkRepository<TKey, TEntity, TDbContext> : IKirelGenericEntityFrameworkRepository<TKey, TEntity> 
+    where TEntity : class, ICreatedAtTrackedEntity, IKeyEntity<TKey> 
+    where TKey :  IComparable, IComparable<TKey>, IEquatable<TKey>
+    where TDbContext : DbContext
+{
+    private readonly DbContext _dbContext;
```

Comment:
You forget about changing type of dbcontext, will be TDbContext

Reply:
Fixed